internal/rbd/rbd_attach.go

```diff
@@ -384,7 +382,7 @@ func detachRBDImageOrDeviceSpec(
 	ctx context.Context,
 	imageOrDeviceSpec string,
 	isImageSpec, isNbd, encrypted bool,
-	volumeID, unmapOptions string) error {
+	volumeID, unmapOptions, logDir string) error {
```

Comment:
This really gets ugly, if passing an existing structure is not practical, create a new `detachRBDImageArgs` with all the needed parameters. That makes calling this function so much simpler to understand.

Reply:
Done now, PTAL.